axon-reactor/src/main/java/org/axonframework/extensions/reactor/commandhandling/gateway/DefaultReactorCommandGateway.java

```diff
@@ -126,6 +124,10 @@ public Registration registerResultHandlerInterceptor(
                 .map(it -> (R) it.getPayload());
     }
 
+    private Mono<? extends CommandResultMessage<?>> mapExceptionalResult(CommandResultMessage<?> response) {
```

Comment:
Wow, interesting. But for Commands I did get it to work with an `onErrorMap`. 
How come this is still necessary?

Reply:
Hm, that's true, we are already doing the same thing in ReactorCallback, so I either keep ReactorCallback unwrapping errors or keep the new `mapExceptionalResult`. What do you think?

Reply:
Hmm though one. My gut would tell me to go for consistency, thus have both do the same for clarity to users/the team. On the other hand, that would entirely drop the need of the `ReactorCallback`, right?

Reply:
`ReactorCallback` would still be needed,  but `isExceptional` check inside would not

Reply:
Gotcha, thanks for that. Then I'd remove the `isExceptional` check from the callback, in favor of consistency between the gateway implementations on this level.